index.html

```diff
@@ -0,0 +1,571 @@
+  - If `name` exists within `namespaces`, the ***Name Uniqueness* validation fails**.
+  - Insert `name => d` into `namespaces`
+- **Return** `namespaces`
+
+**Explanatory Text**
+
+Prefixes, whether implicit or explicit, must be unique within a document. Valid:
+
+<view-of src=prefixing.graphql#schema[0]
+  title="Example of unique prefixes"
+></view-of>
+
+It is also valid to reference multiple versions of the same spec under different prefixes:
+
+<view-of src=prefix-uniqueness.graphql#schema[0]
+  title="Explicit prefixes allow multiple versions of the same spec to coexist within a Document"
+></view-of>
+
+Without the explicit `as:`, the above would be invalid:
+
+<view-of src=prefix-uniqueness.graphql#schema[1]
+  title="Counter-example of non-unique prefixes with multiple versions of the same spec"
+  class=counter-example
+></view-of>
+
+Different specs with the same prefix are also invalid:
+
+<view-of src=prefix-uniqueness.graphql#schema[2]
+  title="Counter-example of non-unique prefixes"
+  class=counter-example
+></view-of>
+
+### Assign Namespaces
+
+Create a map of `element: Any Named Element -> featureName: String | null`, associating every named schema element within the document with the name of a feature, or `null` if it is not associated with a feature.
+
+**Formal Specification**
+- Given [`namespaces`](#namespace-collection-validates-name-uniqueness)
+- Let `assignments` be a map of `element: Any Named Element -> featureName: String | null`, initally empty
+- For each named schema element `e` within the Document,
+  - If `e`'s name begins with `"__"`,
+    - Insert `e => null` into `assignments`
+    - **Continue to next `e`**
+  - If `e`'s name contains the substring `"__"`,
+    - Partition `e`'s name into `[prefix, base]` at the first `"__"` (that is, find the shortest `prefix` and longest `base` such that `e`'s name `= prefix + "__" + base`)
+    - If `prefix` exists within `namespaces`, insert `e => namespaces[prefix]` into `assignments`
+      - Else, insert `e => null` into `assignments`
+    - **Continue to next `e`**
+  - Insert `e => null` into `assignments`
+- **Return `assignments`**
+
+### Is Exported?
+
+Determine if any schema element is [exported](#the-export-schema). A core schema Document's [API](#the-export-schema) is the subset of the entire Document containing only exported elements.
+
+**Formal Specification**
+- Given [`assignments`](#assign-namespaces)
+- Given [`coreName`](#bootstrapping-validates-has-core-feature)
+- Given some schema element `e` within the Document,
+  - For each Directive `d` (if any) on `e`,
+    - If `d`'s name is *coreName*`__export`,
+      - If `d` does not have an `isExport:` argument *or* `isExport:` is `true`, **Return `true`**
+      - If `d` has an `isExport:` argument whose value is `false`, **Return `false`**
```

Comment:
What should we do if it has an `isExport:` argument that is not `true` or `false` (perhaps an explicit null, perhaps something else entirely because the argument has a non Boolean type)?  Is that an error, or is that the same as no `isExport`?

Reply:
I assumed that an earlier GraphQL validation would catch that. Should we make it explicit that such validations must have already run?